client-lite/src/config/mcc_manager.cpp

```diff
@@ -48,57 +55,80 @@ std::string MCCManager::NextHost()
         }
     }
 
+    const auto originalHost = web::uri{originalUrl}.host();
     if (!mccHostName.empty())
     {
-        if (_banList.IsBanned(mccHostName))
-        {
-            _hosts.erase(mccHostName);
-            mccHostName.clear();
-        }
-        else
+        auto it = _mccHostsByOriginalHost.find(originalHost);
+        if (it != _mccHostsByOriginalHost.end())
         {
-            // Record the time of when we first handed out this host
-            if (_hosts.find(mccHostName) == _hosts.end())
+            if (it->second.Address() != mccHostName)
             {
-                _hosts[mccHostName] = std::chrono::steady_clock::now();
+                // MCC to be used has changed, discard existing one
+                _mccHostsByOriginalHost.erase(it);
+                it = _mccHostsByOriginalHost.end();
             }
         }
+
+        if (it == _mccHostsByOriginalHost.end())
+        {
+            MccHost newEntry{mccHostName};
+            it = _mccHostsByOriginalHost.emplace(originalHost, std::move(newEntry)).first;
+        }
+
+        DO_ASSERT(it != _mccHostsByOriginalHost.end());
     }
-    DoLogVerbose("Returning MCC host: [%s]", mccHostName.data());
+    DoLogVerbose("Returning MCC host: [%s] for original host: [%s]", mccHostName.data(), originalHost.data());
     return mccHostName;
 }
 
-bool MCCManager::NoFallback() const
+// Returns true if MCC host is banned, false otherwise
```

Comment:
Comment applies to IsBanned? and not ReportHostError correct? Consider deleting anyways cause the function signature explains it well enough

Reply:
Yes, will delete. The method changed mid way through the changes :-)